src/esp/assets/RenderAssetInstanceCreationInfo.h

```diff
@@ -44,6 +49,7 @@ struct RenderAssetInstanceCreationInfo {
   Corrade::Containers::Optional<Magnum::Vector3> scale;
   Flags flags;
   std::string lightSetupKey;
+  std::shared_ptr<physics::ArticulatedObject> rig;
```

Comment:
RenderAssetInstanceCreationInfo is meant to be a lightweight object that can be easily serialized (JsonEspTypes.h), so I'm a little confused to see this member here. How does it get serialized?

Reply:
I agree that this is not ideal.

I put the rig there because it's a straightforward way to pass it down the resource manager with minimal changes. It does not get serialized like other fields.

I'm not sold on a solution. In the spirit of making this gfx-replayable, perhaps this could be a rig ID that would be independently registered in the resource manager, _and eventually_ match a rig representation in the batch renderer.

What do you think?

Reply:
Yeah. Let's just assume this will get revisited when you integrate skinning into gfx-replay. I'm happy for this to merge as-is.